updated descriptions in properties

lzehl opened the pull request:
updating descriptions for all properties

jcolomb commented:
see discussions about author list schema here https://github.com/pandoc/scholarly-metadata/blob/master/article-authoring-schema.json

lzehl commented:
If we keep affiliations in openMINDS they will be most likely defined in a separate schema, because one affiliation should be valid for multiple persons. Due to the current situation with GDPR it's still an open discussion though how we handle these information.

lzehl commented:
Thanks for your input here, though. Let's keep in touch about this.

jcolomb commented:
I never thought GDPR rules would apply to the researchers in this case... One way to handle this might be to ask only for orcid ID (other information can be obtained from there on demand, and orcid would deal with GDPR rights of the users) ?

skoehnen commented:
This could be something we should bring up in the Data Working Group Meeting. Maybe the DPO already has some thoughts about this.

skoehnen commented:
What should we with this PR, are the issues already resolved?
The I would suggest we close it (The discussion will still be accessible).

lzehl commented:
I'll merge it in to not forget about the issue when doing the v3 schema rewrite. The GDPR issue with this addition is most likely solved in the new Agreements for using the EBRAINS/HBP platform that are currently put together. For v3 I do not see why we can not collect affiliation information for the authors.

Commit:
updated descriptions in properties (first test)

Commit:
Automated vocabulary extraction

Commit:
Update properties.json

Commit:
Update properties.json

Commit:
Update properties.json

Commit:
Update properties.json

Commit:
Update properties.json

Commit:
solved conflict that occurred when  merging in v1

apdavison commented on vocab/properties.json:
```diff
@@ -11,7 +11,7 @@
     },
     "abstractionLevel": {
-        "description": "The abstraction level indicates the extend of simplification of physical, spatial, or temporal details or attributes in the study of objects or systems.",
+        "description": "Extend of simplification of physical, spatial, or temporal details or attributes in the study of objects or systems.",
```
"Extent"

lzehl replied:
@apdavison thanks ! I'll correct it.

apdavison commented on vocab/properties.json:
```diff
@@ -22,7 +22,7 @@
     },
     "accessibility": {
-        "description": null,
+        "description": "Ability to access something.",
```
isn't this the level of accessibility (not just true/false) ?

lzehl replied:
@apdavison : that's a good point. I'll change it.

lzehl commented:
@apdavison there is more to come and I would appreciate your feedback on all descriptions. I find it quite tough to write them to be honest. Would you continue the review?

Commit:
first descriptions for all properties

Commit:
Update properties.json

apdavison commented on vocab/properties.json:
```diff
@@ -73,7 +73,7 @@
     },
     "age": {
-        "description": null,
+        "description": "Time of life or existance at which some particular qualification, capacity or event arises.",
```
existence

lzehl replied:
oh... yes thanks. I'll correct it

apdavison commented on vocab/properties.json:
```diff
@@ -147,15 +147,15 @@
     },
     "author": {
-        "description": null,
+        "description": "Writer or creator of a literary or creative work.",
```
don't we also use it for contributors to a Dataset? (not a literary work, only a creative work in a broad sense)

lzehl replied:
How about adding "... Also includes creators of a dataset publication." or "Writer or creator of a literary or creative work, as well as a dataset publication." ?

UlrikeS91 replied:
https://schema.org/Dataset is listed under "creativeWork". I don't think that we need to specifically mention datasets here.

apdavison commented on vocab/properties.json:
```diff
@@ -584,22 +578,22 @@
     },
     "email": {
-        "description": null,
+        "description": "Means or system for transmitting messages electronically.",
```
I think it is used more specifically to refer to an e-mail _address_

lzehl replied:
Should we also be more specific in the property name? ("emailAddress")
For the description, how about "Address to which an electronic mail can be sent."

UlrikeS91 replied:
I think you followed https://schema.org/email, where the property name is also only "email" (expecting an email address). An email address can be used for both sending and receiving emails. 

Maybe something like: "An address identifying an electronic mail box to which messages are delivered or sent from."

apdavison commented on vocab/properties.json:
```diff
@@ -584,22 +578,22 @@
     },
     "endDate": {
-        "description": null,
+        "description": "Point at which something terminates in time",
```
if it's a date, it's not a point in time. How about "Date on which something terminates"?

lzehl replied:
I see your point. I was trying to follow the rule that the description should explain the terms used for the property and not reuse those terms in the explanation, but in some cases this might be too much and difficult to still be accurate. I'll change it to your suggestion.

apdavison commented on vocab/properties.json:
```diff
@@ -631,7 +625,7 @@
     },
     "fullDocumentation": {
-        "description": null,
+        "description": "Usually printed instructions, comments, and information for using a particular product.",
```
I suggest removing "Usually printed". Maybe replace with "Non-abridged"

lzehl replied:
Yes. Good idea. I'll change it

apdavison commented on vocab/properties.json:
```diff
@@ -735,7 +729,7 @@
     },
     "hasAlternativeVersion": {
-        "description": null,
+        "description": "Reference to a particular form of something differing only in the format, but not the actual content.",
```
I don't think this description is correct. Generally it is the content that differs between versions. Sometimes the format differs, but not usually.

lzehl replied:
This will be our specific definition for "alternative" versions, but I agree that "format" not necessarily covers it. For me an alternative version provides essentially the same data but presented in a different form or context. E.g. one dataset was delivered in a custom structure, and then they present it again BIDS conform. Or one delineation set was presented in reference space A, and a second time in reference space B. 

Do you have an idea how to describe this best?

UlrikeS91 replied:
Suggestion: "Reference to an alternative form of something where the essence of the original form remains unchanged. "

apdavison commented on vocab/properties.json:
```diff
@@ -779,7 +773,7 @@
     },
     "hasParent": {
-        "description": null,
+        "description": "Reference to the source object or legal person.",
```
"source" is rather vague here.

lzehl replied:
I know... you have an idea how to make it more concrete but explaining "parent" in this context?

UlrikeS91 replied:
Maybe stupid suggestion: "Reference an object or legal person to which this object or legal person is the subsidiary to."

lzehl replied:
Changed it to: "Reference to a parental object or legal person."

apdavison replied:
"parent object". "parental" breaks the metaphor, I would only use that for an actual human relationship.

lzehl replied:
@apdavison thanks! I'll changed it.

apdavison commented on vocab/properties.json:
```diff
@@ -791,14 +785,14 @@
     },
     "hasResearchProducts": {
-        "description": null,
+        "description": "Reference to appendent research products.",
```
"appendent" seems to be a legal term not normally used in everyday or scientific English. Would "subsidiary" work?

lzehl replied:
Here I'll trust you fully. I'll change it to "subsidiary"

apdavison commented on vocab/properties.json:
```diff
@@ -840,7 +828,7 @@
     },
     "hash": {
-        "description": null,
+        "description": "Function that converts original data into a single value.",
```
also refers to the resulting single value. ("I used a hash function to calculate the hash of the file")

lzehl replied:
I think we decided to use "digest" (hash digest) for the resulting single value and use "hash" for referring to both (function + digest)? Any idea how to solve this without confusing people too much?

lzehl replied:
And you are right. Currently the description also only covers the "function". How about:
"Term used for the process of converting any data into a single value. Often also directly refers to the resulting single value."

apdavison commented on vocab/properties.json:
```diff
@@ -865,7 +853,7 @@
     },
     "homepage": {
-        "description": null,
+        "description": "Central website of something or someone.",
```
Maybe "Principal" or "Main" rather than "Central"?

lzehl replied:
Whatever you think is better ;). If you don't object I'll use "main"

apdavison commented on vocab/properties.json:
```diff
@@ -911,23 +899,23 @@
     },
     "identifierSchema": {
-        "description": null,
+        "description": "Service institution providing particular formatted identifier for something or someone.",
```
I don't really understand this one.

lzehl replied:
It should describe e.g. "DOI", "ORCID", "UUID" as a service. But I guess it's rather a standard?

lzehl replied:
Would the following be better?
"Standard for creating a particular identifier for something or someone."

Commit:
update based on review request

two corrections are still missing:
+ better description for "hasAlternativeVersion"
+ better description for "hasParent"

lzehl commented:
@apdavison thanks for providing such a quick review on this. I've updated most of what you noticed (and hopefully did not introduce more typos ;)), except for the description of "hasAlternativeVersion", and "hasParent". Do you have ideas for those?

Commit:
Update properties.json

Commit:
Update properties.json

lzehl commented:
@apdavison except for the description for "hasParents" I've updated now all. Could you review the changes and see if they are correct? Thanks a lot in advance!

UlrikeS91 commented:
@lzehl: All property name changes in the schemas are done through PRs now (still need to be reviewed and merged). I forgot to note done anywhere that we want to update the names of the properties to reflect that potential plural, e.g.

```
    "author": {
        [...]
        "name": "Author",
        [...]  
    },
```

to

```
    "author": {
        [...]
        "name": "Author(s)",
        [...]  
    },
```

I didn't want to start this myself before 1) the properties.json is updated with the property changes from the different PRs and 2) your PR here is merged/closed.

Commit:
updated properties according to the latest changes in the openMINDS submodules

lzehl commented:
@apdavison : I've updated the properties according to the latest changes in the openMINDS submodules. All deprecated properties should be removed (hope I did not overlook one) and all existing properties should have a description now. Could you review it again?

@UlrikeS91 : thanks for the reminder, but when I went over it now, I realized that we can't do this. The name is applied to all schemas that use this property, but not all schemas do expect the same number of values for that property. So we have to keep this information (how many values are you allowed to enter) in the instructions.

apdavison commented:
@lzehl I'm a bit over-committed this week and next, so please don't wait for my review.

lzehl commented:
@apdavison okay, I can ask @UlrikeS91 to have a final check and to merge it.

UlrikeS91 commented on vocab/properties.json:
```diff
@@ -148,15 +148,8 @@
-    },
     "author": {
-        "description": null,
+        "description": "Creator of a literary or creative work, as well as a dataset publication.",
```
https://schema.org/Dataset is listed under "creativeWork". I don't really think that we need to specifically mention datasets here.

lzehl replied:
This is true for schema.org, but actually not true from the legal perspective (at least if you focus on the data... only the publication part, meaning the metadata presentation in a database and data descriptors are considered as creative work in the legal context as far as I understand these matters). Let's leave the specific description for now

UlrikeS91 replied:
That makes sense :)

UlrikeS91 commented on vocab/properties.json:
```diff
@@ -22,7 +22,7 @@
     },
     "accessibility": {
-        "description": null,
+        "description": "Level of ability to access something.",
```
Maybe: "The degree to which something is available to someone or something."

lzehl replied:
Available is not the same as accessibility. But I like the rest of what you suggested. I've entered now a mix:
"Level to which something is accessible to someone or something."

> I've added some suggestions for some of the descriptions. Some of them need to be changed because the description is wrong (e.g. handedness).

Handedness is not wrong... https://www.merriam-webster.com/dictionary/handedness

And you did not commented on it explicitly... was this a typo?

UlrikeS91 replied:
I think you hopped over a few of my comments. There is a comment related to "handedness" directly. I'm sorry, I could have been a bit more specific with my main comment about this. It's not that the description is wrong per se, but in our context with the options available, it's more of a gradient (similar to what Andrew commented for "accessibility"). Our options are: left, right, both equally good, different one per task, both equally bad (was too lazy to look up the exact words). 
Therefore, the description "Tendency to use one hand rather than the other." is not sufficient. Also, handedness applies to feet as much as to hands, right? 
I also commented on the description for "laterality" (description: "Dominance in function or use of one of a pair of lateral homologous parts of the body."). In our context, this is not really true for "laterality" (anymore; since we split handedness and laterality now anyway) but this description would already be much better for handedness if you don't like my suggestions under handedness ;)

UlrikeS91 replied:
The description for accessibility is nice :)

UlrikeS91 commented on vocab/properties.json:
```diff
@@ -511,7 +485,7 @@
     },
     "developer": {
-        "description": null,
+        "description": "Legal person that develops computer software.",
```
This is too specific. A legal person can develop more than just a computer software to be a developer. Relevant definitions I found:
IT related:
"A person or company that creates new products, especially software, or services."
"A developer is a person who creates something with a computer."
"A person or company that develops software, applications, etc."
A bit more general:
"A person or entity engaged in the creation or improvement of certain classes of products."

My suggestion would be: "A legal person that creates or improves products or services (e.g., software, applications, etc.)."

lzehl replied:
fine with me although we purely use it in the software context.

UlrikeS91 replied:
Not quite. We have developers also for models (in fact, a required field for "modelVersion").

UlrikeS91 commented on vocab/properties.json:
```diff
@@ -595,22 +569,22 @@
     },
     "email": {
-        "description": null,
+        "description": "Address to which an electronic mail can be sent.",
```
Stick to property name "email" (related to the exchange you had with @apdavison about this). I think you followed https://schema.org/email, where the property name is also only "email" (expecting an email address). 

I'm not 100% happy with the description. An email address can be used for both sending and receiving emails.

Maybe something like: "An address identifying an electronic mail box to which messages are delivered or sent from."

lzehl replied:
Changed it to : "Address to which or from which an electronic mail can be sent."

UlrikeS91 commented on vocab/properties.json:
```diff
@@ -595,22 +569,22 @@
     },
     "endDate": {
-        "description": null,
+        "description": "Date at which something terminates in time",
```
based on @apdavison & @lzehl comments, maybe: "A particular day, month and year on which something is terminated or completed." Alternatively, since avoiding date is a bit tricky: "A particular calendar date on which something is terminated or completed." ("calendar date" should cover the "in time"-portion; "A particular" could also be removed)

lzehl replied:
I leave it as it is. It's not necessary to further define it I think. If we want to be precise it would be "Date in the Gregorian calendar at which..." But I think that's over the top here.

UlrikeS91 commented on vocab/properties.json:
```diff
@@ -621,15 +595,15 @@
     },
     "familyName": {
-        "description": null,
+        "description": "Name borne in common by members of a family.",
```
Maybe: "The portion of a person's personal name that indicates their family, tribe or community."

lzehl replied:
I think the given description is sufficient. Note that I usually try to stay close to what other sources use as definition: https://www.merriam-webster.com/dictionary/surname (second meaning)

UlrikeS91 commented on vocab/properties.json:
```diff
@@ -734,15 +708,15 @@
     },
     "givenName": {
-        "description": null,
+        "description": "Name given to a person, including all potential middle names, but excluding the family name.",
```
Maybe: "The portion of a person's personal name, including all potential middle names, that differentiates that person from other members of a group who have a common family name."

lzehl replied:
I don't see that this is giving more information, in particular there are cultures where there is no family name.

UlrikeS91 commented on vocab/properties.json:
```diff
@@ -639,14 +613,14 @@
     },
     "format": {
-        "description": null,
+        "description": "Method of organizing data.",
```
Maybe: "The way in which information or data is arranged and stored on a computer according to a chosen pattern." (alternatively without "on a computer")

lzehl replied:
Also here I've sticked close to https://www.merriam-webster.com/dictionary/format (third meaning). Note also, stating the general meaning is fine if we also use this word in the general sense. If we use the word in a more specific meaning we should choose the description accordingly. It is our namespace description therefore we can be specific in what we mean. Nonetheless in some cases we want the general meaning because we do use or may want to use the word in the general sense.

UlrikeS91 commented on vocab/properties.json:
```diff
@@ -753,7 +727,7 @@
     },
     "handedness": {
-        "description": null,
+        "description": "Tendency to use one hand rather than the other.",
```
Handedness isn't just left/right preference. This needs to be changed.

My suggestion: "Degree of dominance exhibited by an organism in the use of one hand or foot in comparison to the other hand or foot in the performance of tasks." (dominance could be replaced by preference)

lzehl replied:
I agree here that your definition is more general and matches more what we cover. I would like to shorten it though: "Degree to which an organism prefers one hand or foot over the other hand or foot during the performance of a task."

UlrikeS91 commented on vocab/properties.json:
```diff
@@ -1032,7 +975,7 @@
     },
     "isAlternativeVersionOf": {
-        "description": null,
+        "description": "Reference to an original form which essence was unchanged, but presented in an alternative form.",
```
This is very hard to understand... 

Our original suggestion was "Reference to an alternative form of something where the essence of the original form remains unchanged.". I guess you changed it to be more consistent within the property list here. 

Maybe: "Reference to a particular form of something that was altered without changing its essence."

I still prefer our orginial suggestion, though.

lzehl replied:
it's the wrong terminus though... You refer to the original. And the changed one is the one that holds the property. Correctly but from my perspective similarly awkward to read would be "Reference to an original form of something that was then in its form without changing its essence." any better idea? With those properties it is quite hard to not mix it with the instruction...

UlrikeS91 commented on vocab/properties.json:
```diff
@@ -1107,7 +1050,7 @@
     },
     "laterality": {
-        "description": null,
+        "description": "Dominance in function or use of one of a pair of lateral homologous parts of the body.",
```
This needs to be changed. The description fits the term, yes, but  it's not describing the term in the way we use it...
This goes back to the general issue of using "laterality" to describe which part of a bilateral something has been used (left, right or both). 

I cannot come up with a good suggestion here. 

This description could also be used to describe handedness if you don't like my suggestion there ;)

lzehl replied:
In my opinion it does, because "function or use" can be interpreted in many different ways. E.g. "function" can mean the left kidney works better than your right. Or the left hemisphere is more responsible for something than the right. "Use" can mean that the left hemisphere was used not the right. And the portion of left vs right hand in handedness reflects the laterality of handedness. So it fits there as well, although we are more specific in handedness now.

UlrikeS91 commented on vocab/properties.json:
```diff
@@ -1386,7 +1327,7 @@
+        "description": "Manner, position, or direction in which physical properties are set to determine the function, characteristics or behavior of something.",
         "linkedTypes": [
             "https://openminds.ebrains.eu/core/ParameterSetting"
         ],
```
Just as a note: This will become deprecated once PR https://github.com/HumanBrainProject/openMINDS_core/pull/177 has been merged. 
Renamed to "parameterSet". I'm not sure if the description for parameterSetting is quite suitable for the new property name...

lzehl replied:
It does not really change the meaning.... But one could be a bit more specific in our context: "Manner, position, or direction in which physical or digital properties are set to determine a particular function, characteristics or behavior of something." what do you think? I need to merge in the main branch again or correct this in a second PR

UlrikeS91 replied:
Sounds good

UlrikeS91 commented on vocab/properties.json:
```diff
@@ -1689,14 +1624,14 @@
     },
     "startDate": {
-        "description": null,
+        "description":"Date at which something begins in time",
```
related to comment in line 580 ("endDate" --> "A particular day, month and year on which something is terminated or completed." OR "A particular calendar date on which something is terminated or completed.")

This needs to be changed accrodingly to either: "A particular day, month and year on which something began." OR "A particular calendar date on which something began."

UlrikeS91 commented on vocab/properties.json:
```diff
@@ -1720,7 +1655,7 @@
     },
     "studiedState": {
-        "description": null,
+        "description": "Reference to a point in time and condition in which something or someone was studied.",
```
Maybe: "Reference to a mode or condition of being at the point of examination"

lzehl replied:
That's not true. The time is actually the essential keyword here. Condition could be even deleted because that is what is depending on the time. What about:

"Reference to a point in time at which something or someone was studied in a particular mode or condition." ??

UlrikeS91 replied:
Sounds good :)

UlrikeS91 commented on vocab/properties.json:
```diff
@@ -1794,7 +1728,7 @@
     },
     "terminology": {
-        "description": null,
+        "description": "Nomenclature for a particular field of study.",
```
Rather: "A body of terms used in relation to a particular field."

lzehl replied:
I want to leave it. "nomenclature" is quite fitting (cf. https://www.merriam-webster.com/dictionary/nomenclature)

UlrikeS91 commented on vocab/properties.json:
```diff
@@ -1874,7 +1808,7 @@
     },
     "versionInnovation": {
-        "description": null,
+        "description": "Documentation on what change compared to an older version of something.",
```
Maybe: "Documentation on what changed in comparison to a previous particular form of something."

lzehl replied:
I've changed it to: "Documentation on what changed in comparison to a previously published form of something."

UlrikeS91 commented on vocab/properties.json:
```diff
@@ -1898,22 +1832,22 @@
     },
     "webpage": {
-        "description": null,
+        "description": "Hypertext document (block of information) found at the World Wide Web.",
```
"Hypertext document (block of information) found **ON** the World Wide Web."

lzehl replied:
yup. that's an update error... previously defined it as "Hypertext document (block of information) found at a single World Wide Web address."

UlrikeS91 reviewed (changes requested):
I've added some suggestions for some of the descriptions. Some of them need to be changed because the description is wrong (e.g. handedness).

Commit:
Update properties.json

Commit:
Update properties.json

Commit:
Update properties.json

Commit:
final update for PR
